aiida/orm/nodes/node.py

```diff
@@ -47,9 +49,16 @@ def __init__(self, fileobj, name):
     def _warn_if_not_entered(self, method):
         """Fire a warning if the object wrapper has not yet been entered."""
         if not self._was_entered:
-            msg = '`{}` used without context manager for {}. This will raise starting from `aiida-core==2.0.0`'.format(
-                method, self._name
-            )
+            msg = f'\n`Node.{method}` used outside context manager for {self._name}.\n' + \
```

Comment:
The `Node.{method}` is not actually correct and may be misleading. The `method` here is what is actually called on the *return value* of `Node.open`. Normally this is a filelike object, but now it is this instance of `WarnWhenNotEntered` that we added. You can see this when this object is garbage collected, because it will then say `Node.del used outside ..`  but `Node` doesn't have a `del` method. Same goes for `read` though.

Consider following example:
```
handle = node.open()
content = handle.read()
```
It is when `read` is called on `handle` that we get the warning, not on the node. Likewise, it is `del handle` that is called when it is garbage collected that incurs the second warning.

This shows another potential problem, with the example above, the line of code that you will include in your warning will be `content = handle.read()` and people may be confused by you to tell them that this should be `with node.open()`. Note that in this case, the fix may be obvious, because they will go to the line mentioned, and see the `node.open()` call straight above it. However, they are not necessarily very close. They could have passed the handle to some function, so then it may not be clear at all where the actual problem is.

Reply:
> The `Node.{method}` is not actually correct and may be misleading. The `method` here is what is actually called on the _return value_ of `Node.open`.

Right, I've now tried to rephrase the message to make this clearer. Further suggestions welcome.

> the line of code that you will include in your warning will be `content = handle.read()` and people may be confused by you to tell them that this should be `with node.open()`

You are very welcome to suggest a better way.
In my view, this warning already improves the situation enormously, since it shows which call caused the warning rather than where the warning is defined.
For me this is good enough.